websocket/transceiver/transceiver_test.go

```diff
@@ -117,15 +117,12 @@ func TestReceiver_ListenAndSendAck(t *testing.T) {
 		},
 	})
 
-	wg := &sync.WaitGroup{}
-	wg.Add(1)
-
 	count := 0
 	conn := &testscommon.WebsocketConnectionStub{
 		ReadMessageCalled: func() (int, []byte, error) {
 			time.Sleep(500 * time.Millisecond)
-			if count >= 1 {
-				wg.Done()
+			if count == 2 {
```

Comment:
is this ok? Should you use a mutex to access the counter value?

Reply:
refactored